services/AUTOMATIC1111/entrypoint.sh

```diff
@@ -8,8 +8,22 @@ mkdir -p /data/config/auto/scripts/
 find "${ROOT}/scripts/" -maxdepth 1 -type l -delete
 cp -vrfTs /data/config/auto/scripts/ "${ROOT}/scripts/"
 
+# config.json for auto
+## Step 1: Copy default config to user config if missing
 cp -n /docker/config.json /data/config/auto/config.json
-jq '. * input' /data/config/auto/config.json /docker/config.json | sponge /data/config/auto/config.json
+## Step 2: Put the default config and user config into vars
+cfg_json_default=$(cat /docker/config.json)
+cfg_json_user=$(cat /data/config/auto/config.json)
+## Step 3: delete invalid entries from user.json
+cfg_json_user=$(echo $cfg_json_user | jq '
+  . as $original
+    | reduce (to_entries[] | select((.key | startswith("outdir_"))
+        and (.value | test("^\/output(\\.)?(\/\\.?[\\w\\-\\_]+)+\/?")|not)))
+        as $item
+        ($original; del(.[$item.key]))
+')
+## Step 4: Merge and save config
+echo $cfg_json_default $cfg_json_user | jq '. * input' | sponge /data/config/auto/config.json
```

Comment:
Thank you for putting the effort into this, can we move this into a python script? this works but it is unmaintainable in the long run.

Reply:
it has to be called from the entrypoint, because the config does not exist in the dockerfile (yet)